subprojects/core/src/main/java/org/gradle/execution/plan/TaskNodeFactory.java

```diff
@@ -64,4 +82,55 @@ public void clear() {
         nodes.clear();
     }
 
+    private static class DefaultTypeOriginInspectorFactory {
```

Comment:
I wonder how all of this works when we have a configuration cache hit. I suppose we won't find any plugins, will we?

Reply:
You're right that it probably doesn't work with the configuration cache. I'm not sure we can easily make it work...